tests/components/uptime_kuma/test_config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Tests for the Uptime Kuma config flow."""
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.uptime_kuma.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+    CONTENT_TYPE_JSON,
+)
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+FIXTURE_USER_INPUT = {
+    CONF_HOST: "https://uptimekuma.test.com",
+    CONF_PORT: 443,
+    CONF_USERNAME: "user",
+    CONF_PASSWORD: "pass",
+    CONF_VERIFY_SSL: True,
+}
+
+
+async def test_connection_error(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test we show user form on Uptime Kuma connection error."""
+    aioclient_mock.get(
+        f"{FIXTURE_USER_INPUT[CONF_HOST]}:{FIXTURE_USER_INPUT[CONF_PORT]}",
+        text="",
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=FIXTURE_USER_INPUT
+    )
+
+    assert result
+    assert result.get("type") == data_entry_flow.RESULT_TYPE_FORM
```

Comment:
Could you have a look at your failed tests? It seems you are not mocking all requests correctly.

And I believe it is better to mock your python client functions, instead of the aio http client directly. However, someone else with more testing experience should confirm this.

Reply:
`side_effect` can be used on the mocked object

Reply:
@meichthys ☝️